.gitignore

```diff
@@ -22,3 +22,5 @@ Testing/
 docs/_build/
 
 node_modules
+
+compile_commands.json
```

Comment:
What is this?

Reply:
It is generated by `cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_EXPORT_COMPILE_COMMANDS=1`

The file is used by clangd and other tools to parse the source tree and support function jump. People who use it to read code would want to ignore it as its content is local to the compiling system.

https://clang.llvm.org/docs/JSONCompilationDatabase.html

Reply:
Yes, this should be ignored.